### PR TITLE
📖 Add getting started doc

### PR DESCRIPTION
**Pull request by camilamacedo86**
## Description

This PR introduces an overview or summary to assist users in getting started more smoothly. We've designed this document to address the most frequently asked initial questions.

While the tutorials offer in-depth information, this document aims to provide a broad understanding in a more concise manner.

See the preview: https://deploy-preview-3596--kubebuilder.netlify.app/getting-started

Closes: https://github.com/kubernetes-sigs/kubebuilder/issues/3203

### CODE REVIEW

File: docs/book/src/getting-started.md [hunk 1]
```diff
@@ -0,0 +1,537 @@
...
+
+<aside class="note">
+<h1>Getting a better idea</h1>
+
+Consider a typical scenario where the objective is to run an application and its database on a Kubernetes platform. In this context, one object might represent the App, while another denotes the DB. If we define one CRD for the App and another for the DB, we uphold essential concepts like encapsulation, the single responsibility principle, and cohesion. Breaching these principles might lead to complications, making extension, reuse, or maintenance challenging.
+
+In essence, the App CRD will have the DB CRD as its controller. Let's say, for instance, that the application requires a Deployment and Service to run. In this example, the App’s Controller will cater to these needs. Similarly, the DB’s controller will manage the business logic of its items.
```
Comment (grzesuav) [inline docs/book/src/getting-started.md:45]: > In essence, the App CRD will have the DB CRD as its controller.

not sure I am following here, can you elaborate ?

**Review by grzesuav — APPROVED** [review]

File: docs/book/src/getting-started.md [hunk 2]
```diff
@@ -0,0 +1,537 @@
...
+## Sample Project
+
+We will create a sample project to let you know how it works. This sample will:
+
+- Reconcile a Memcached CR
+- Create a Deployment with the Memcached image
+- Not allow more instances than the size defined in the CR which will be applied
+- Update the Memcached CR status 
+
+Following the steps.
```
Comment (Sajiyah-Salat) [inline docs/book/src/getting-started.md:16]:
```suggestion


###Following the steps.
```

in preview following the steps feels like out of plsce it should br near to create a project.

File: docs/book/src/getting-started.md [hunk 3]
```diff
@@ -0,0 +1,537 @@
+# Getting Started 
+
+## Overview
+
+By following the [Operator Pattern][k8s-operator-pattern], it’s possible not only to provide all expected resources but also to manage them dynamically, programmatically, and at execution time. To illustrate this idea, imagine if someone accidentally changed a configuration or removed a resource by mistake; in this case, the operator could fix it without any human intervention. Before following the next steps ensure that you have the Operator-SDK tool with a version >= 1.0 installed locally. [To install, check its guide](#installation-guide).
```
Comment (Sajiyah-Salat) [inline docs/book/src/getting-started.md:5]:
```suggestion
By following the [Operator Pattern][k8s-operator-pattern], it’s possible not only to provide all expected resources but also to manage them dynamically, programmatically, and at execution time. To illustrate this idea, imagine if someone accidentally changed a configuration or removed a resource by mistake; in this case, the operator could fix it without any human intervention. Before following the next steps ensure that you have the Operator-SDK tool with a version >= 1.0 installed locally. To install, check its [guide](installation-guide).
```

File: docs/book/src/getting-started.md [hunk 4]
```diff
@@ -0,0 +1,537 @@
...
+
+## Sample Project
+
+We will create a sample project to let you know how it works. This sample will:
+
+- Reconcile a Memcached CR
```
Comment (Sajiyah-Salat) [inline docs/book/src/getting-started.md:11]: Please elaborate what is memcached here.

File: docs/book/src/getting-started.md [hunk 5]
```diff
@@ -0,0 +1,537 @@
...
+At this point, you can primarily execute the commands highlighted in the [quick-start][quick-start]. 
+By executing `make build IMG=myregistry/example:1.0.0`, you'll build the image for your project. For testing purposes, it's recommended to publish this image to a 
+public registry. This ensures easy accessibility, eliminating the need for additional configurations. Once that's done, you can deploy the image 
+to the cluster using the `make deploy IMG=myregistry/example:1.0.0` command.
+
+## Next Steps
+
+- To delve deeper into developing your solution, consider going through the provided tutorials.
+- For insights on optimizing your approach, refer to the [Best Practices][best-practices] documentation.
+
```
Comment (Sajiyah-Salat) [inline docs/book/src/getting-started.md:525]: Add a link for installation guid

File: docs/book/src/getting-started.md [hunk 6]
```diff
@@ -0,0 +1,537 @@
...
+	}
+
+	return ctrl.Result{}, nil
+}
+```
+
+#### Observing changes on cluster 
+
+This controller is persistently observant, monitoring any events associated with this Kind. As a result, pertinent changes 
+instantly set off the controller's reconciliation process. It's worth noting that we have implemented the `watches` feature. [(More info)[watches]].
```
Comment (Sajiyah-Salat) [inline docs/book/src/getting-started.md:419]:
```suggestion
instantly set off the controller's reconciliation process. It's worth noting that we have implemented the `watches` feature. [(More info)][watches]. 
```

**Comment by Sajiyah-Salat** [comment]
Hello @camilamacedo86 I have done review for the first time. maybe there are suggestions that are not important. Kindly ignore that. I am learning.

File: docs/book/src/getting-started.md (hunk 2 above)
Comment (camilamacedo86) [reply docs/book/src/getting-started.md:16]: The text following the steps should not be a subsection. It is just a description to let the audience be aware that the steps will be provided bellow.

File: docs/book/src/getting-started.md [hunk 7]
```diff
@@ -0,0 +1,537 @@
...
+
+<aside class="note">
+<h1>Getting a better idea</h1>
+
+Consider a typical scenario where the objective is to run an application and its database on a Kubernetes platform. In this context, one object might represent the App, while another denotes the DB. If we define one CRD for the App and another for the DB, we uphold essential concepts like encapsulation, the single responsibility principle, and cohesion. Breaching these principles might lead to complications, making extension, reuse, or maintenance challenging.
+
+In essence, the App CRD will have the DB CRD as its controller. Let's say, for instance, that the application requires a Deployment and Service to run. In this example, the App’s Controller will cater to these needs. Similarly, the DB’s controller will manage the business logic of its items.
```
Comment (camilamacedo86) [inline docs/book/src/getting-started.md:45]:
```suggestion
In essence, the App CRD and the DB CRD will have their controller. Let's say, for instance, that the application requires a Deployment and Service to run. In this example, the App’s Controller will cater to these needs. Similarly, the DB’s controller will manage the business logic of its items.
```

File: docs/book/src/getting-started.md (hunk 5 above)
Comment (camilamacedo86) [reply docs/book/src/getting-started.md:525]: Here we are adding the Next Steps.
Ideally who is in this point already followed up the Quick Start and this overview, their for the person has the Kubebuilder CLI installed.

**Comment by camilamacedo86** [comment]
HI @Sajiyah-Salat,

Thank you for your review. The suggestions were addressed could you please check it out and LGTM if you are OK with it?

**Comment by Sajiyah-Salat** [comment]
I will check that shortly

File: docs/book/src/getting-started.md (hunk 1 above)
Comment (camilamacedo86) [reply docs/book/src/getting-started.md]: changed thx

File: docs/book/src/getting-started.md (hunk 2 above)
Comment (Sajiyah-Salat) [reply docs/book/src/getting-started.md:19]: image

To me this doesnt looks good. I mean the format. I think we should either add something after following the steps or remove it or make it near create the project. What say?

File: docs/book/src/getting-started.md (hunk 5 above)
Comment (Sajiyah-Salat) [reply docs/book/src/getting-started.md:528]: Got it.

**Comment by Sajiyah-Salat** [comment]
Everything looks good. Just check this [comment](https://github.com/kubernetes-sigs/kubebuilder/pull/3596#discussion_r1322272593) 
LGTM

**Review by grzesuav — APPROVED** [review]

File: docs/book/src/getting-started.md [hunk 8]
```diff
@@ -0,0 +1,540 @@
...
+                maximum: 3 ## See here from the marker +kubebuilder:validation:Maximum=3
+                minimum: 1 ## See here from the marker +kubebuilder:validation:Minimum=1
+                type: integer
+            type: object
+
+```
+
+#### Sample of Custom Resources
+
+The manifests located under the "samples" directory serve as examples of Custom Resources that can be applied to the cluster.
```
Comment (varshaprasad96) [inline docs/book/src/getting-started.md:127]:
```suggestion
The manifests located under the "config/samples" directory serve as examples of Custom Resources that can be applied to the cluster.
```

File: docs/book/src/getting-started.md [hunk 9]
```diff
@@ -0,0 +1,540 @@
...
+// Check if a Service for the app exists, if not, create one 
+// If there's an error, then restart from the beginning of the reconcile
+if err != nil {
+return reconcile.Result{}, err
+}
+
+// Look for Database CR/CRD 
+// Check the Database Deployment's replicas size
+// If deployment.replicas size doesn't match cr.size, then update it
+// Then, restart from the beginning of the reconcile
```
Comment (varshaprasad96) [inline docs/book/src/getting-started.md:167]: Can we add the comment that in this case we Requeue the reconcile request even when there is no error.

File: docs/book/src/getting-started.md [hunk 10]
```diff
@@ -0,0 +1,540 @@
...
+...
+
+// If at the end of the loop:
+// Everything was executed successfully, and the reconcile can stop  
+return reconcile.Result{}, nil
+
+}
+```
+
+#### Return Options
```
[the remark this answers is not in the document]
Comment (camilamacedo86) [reply docs/book/src/getting-started.md:180]: It seems for me that we have not doc in any place all options. 
So, I think that would be great keep this one. WDYT? Can we keep it?

File: docs/book/src/getting-started.md [hunk 11]
```diff
@@ -0,0 +1,540 @@
...
+// Check if a Service for the app exists, if not, create one 
+// If there's an error, then restart from the beginning of the reconcile
+if err != nil {
+return reconcile.Result{}, err
+}
+
+// Look for Database CR/CRD 
+// Check the Database Deployment's replicas size
+// If deployment.replicas size doesn't match cr.size, then update it
+// Then, restart from the beginning of the reconcile
```
Comment (camilamacedo86) [inline docs/book/src/getting-started.md:167]:
```suggestion
// Then, restart from the beginning of the reconcile. For example, by returning `reconcile.Result{Requeue: true}, nil`.
```

**Comment by camilamacedo86** [comment]
@varshaprasad96 

Thank you for your review.
I think I addressed all your comments/suggestions.
WDYT? Could we move with this one?

**Pull request merged**